libs/langchain/langchain/llms/google/generativeai/base.py

```diff
@@ -0,0 +1,110 @@
+"""Google GenerativeAI Attributed Question and Answering (AQA) service.
+
+The GenAI Semantic AQA API is a managed end to end service that allows
+developers to create responses grounded on specified passages based on
+a user query. For more information visit:
+https://developers.generativeai.google/guide
+"""
+
+from typing import Any, List, Optional
+
+from langchain.pydantic_v1 import BaseModel, PrivateAttr
+from langchain.schema.runnable import RunnableSerializable
+from langchain.schema.runnable.config import RunnableConfig
+
+_import_err_msg = (
+    "`google.generativeai` package not found, "
+    "please run `pip install google-generativeai`"
+)
+
+
+class AqaModelInput(BaseModel):
+    """Input to `GenAIAqa.invoke`.
+
+    Attributes:
+        prompt: The user's inquiry.
+        source_passages: A list of passage that the LLM should use only to
+            answer the user's inquiry.
+    """
+
+    prompt: str
+    source_passages: List[str]
+
+
+class AqaModelOutput(BaseModel):
+    """Output from `GenAIAqa.invoke`.
+
+    Attributes:
+        answer: The answer to the user's inquiry.
```

Comment:
Might want to document somewhere that the answer may be blank if we could not answer the query.